bundle/edu.gemini.ags/src/main/scala/edu/gemini/ags/impl/MultiProbeStrategy.scala

```diff
@@ -0,0 +1,128 @@
+package edu.gemini.ags.impl
+
+import edu.gemini.ags.api.AgsMagnitude.MagnitudeTable
+import edu.gemini.ags.api.AgsStrategy.Estimate
+import edu.gemini.ags.api.{AgsAnalysis, AgsGuideQuality, AgsMagnitude, AgsStrategy}
+import edu.gemini.catalog.api.CatalogQuery
+import edu.gemini.skycalc.Angle
+import edu.gemini.spModel.ags.AgsStrategyKey
+import edu.gemini.spModel.core.SiderealTarget
+import edu.gemini.spModel.guide.{GuideProbe, ValidatableGuideProbe}
+import edu.gemini.spModel.obs.context.ObsContext
+import edu.gemini.spModel.obscomp.SPInstObsComp
+import edu.gemini.spModel.telescope.{PosAngleConstraint, PosAngleConstraintAware}
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+import scalaz._
+import Scalaz._
+
+/**
+ * Represents an AGS strategy consisting of multiple substrategies acting together.
+ */
+sealed case class MultiProbeStrategy(key: AgsStrategyKey, strategies: List[AgsStrategy]) extends AgsStrategy {
+
+  override def magnitudes(ctx: ObsContext, mt: MagnitudeTable): List[(GuideProbe, AgsMagnitude.MagnitudeCalc)] =
+    strategies.flatMap(_.magnitudes(ctx, mt))
+
+  override def analyze(ctx: ObsContext, mt: MagnitudeTable): List[AgsAnalysis] =
+    strategies.flatMap(_.analyze(ctx, mt))
+
+  override def analyze(ctx: ObsContext, mt: MagnitudeTable, guideProbe: ValidatableGuideProbe, guideStar: SiderealTarget): Option[AgsAnalysis] =
+    AgsAnalysis.analysis(ctx, mt, guideProbe, guideStar, probeBands)
+
+  override def catalogQueries(ctx: ObsContext, mt: MagnitudeTable): List[CatalogQuery] =
+    strategies.flatMap(_.catalogQueries(ctx, mt))
+
+  override def candidates(ctx: ObsContext, mt: MagnitudeTable): Future[List[(GuideProbe, List[SiderealTarget])]] =
```

Comment:
I think this can return the same guide probe multiple times. Can the upstream handle that case, or should we add the targets for the same guide probe?